README.md
## Gradient Textfield
 A customizable gradient Textfield for the developers who wants to style the Texfield of Flutter App.
 
 ## Installation
 
 1. Add the current version of package to your pubspec.yaml 
 
 ```dart
 dependencies:
  gradient_textfield: ^0.0.1
  ```
2. Run the following command
 ```dart
flutter pub get
  ```
 3. Import the custom_field package and use it in your Flutter app
  ```dart
import 'package:gradient_textfield/gradient_textfield.dart';
  ```
 ## Example
 The properties you can modify are listed below
 - height
 - width
 - colors (graident)
 - text
 - radius
 - font style
 ## Sample Code

```dart
 class Login extends StatelessWidget {
  const Login({ Key? key }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    TextEditingController email = TextEditingController();
    return Scaffold(
      body:  Gradienttextfield(
              controller: email,
              radius: 40,
              height: 60,
              width: 400,
              colors: const [Colors.grey, Colors.white],
              text: "Email",
              fontColor: Colors.black,
              fontSize: 15,
              fontWeight: FontWeight.normal,
            ),
      
    );
  }
}
```
## Screenshot
<img width="263" alt="package_image" src="https://user-images.githubusercontent.com/54928117/150742278-689d7c37-7273-407b-b414-d89d7364624d.png">



## Upcoming releases
- [ ] Set prefix icons
- [ ] Set Label text and label style
- [ ] Implementing shadows to Containers

 


CHANGELOG.md
## [0.0.1] - Release

* A customizable gradient Textfield for the developers who concentrates on UI writtern in dart.


lib/src/customtextfield.dart
import 'package:flutter/material.dart';

class Customtextfield extends StatefulWidget {
  Customtextfield(
      {Key? key,
      required this.height,
      required this.width,
      required this.colors,
      required this.text,
      this.radius,
     required this.controller,
      this.fontColor,
      this.fontWeight,
      this.fontSize})
      : super(key: key);

  double? height;
  double? width;
  List<Color> colors;
  String? text;
  double? radius;
  TextEditingController? controller;
  Color? fontColor;
  FontWeight? fontWeight;
  double? fontSize;

  @override
  _CustomtextfieldState createState() => _CustomtextfieldState();
}

class _CustomtextfieldState extends State<Customtextfield> {
 
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(left: 20.0, right: 20),
      child: Container(
        height: widget.height,
        width: widget.width,
        decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: widget.colors,
            ),
            borderRadius: BorderRadius.circular(widget.radius ?? 20.0)),
        child: TextField(
          controller: widget.controller,
          decoration: InputDecoration(
              hintText: widget.text,
              hintStyle: TextStyle(
                  color: widget.fontColor ?? Colors.black,
                  fontSize: widget.fontSize ?? 18,
                  fontWeight: widget.fontWeight ?? FontWeight.bold),
              enabledBorder: const OutlineInputBorder(
                  borderSide: BorderSide(color: Colors.transparent)),
              focusedBorder: const OutlineInputBorder(
                  borderSide: BorderSide(color: Colors.transparent))),
        ),
      ),
    );
  }
}


lib/src/gradienttextfield.dart
import 'package:flutter/material.dart';

class Gradienttextfield extends StatefulWidget {
  Gradienttextfield(
      {Key? key,
      required this.height,
      required this.width,
      required this.colors,
      required this.text,
      this.radius,
     required this.controller,
      this.fontColor,
      this.fontWeight,
      this.fontSize})
      : super(key: key);

  double? height;
  double? width;
  List<Color> colors;
  String? text;
  double? radius;
  TextEditingController? controller;
  Color? fontColor;
  FontWeight? fontWeight;
  double? fontSize;

  @override
  _GradienttextfieldState createState() => _GradienttextfieldState();
}

class _GradienttextfieldState extends State<Gradienttextfield> {
 
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(left: 20.0, right: 20),
      child: Container(
        height: widget.height,
        width: widget.width,
        decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: widget.colors,
            ),
            borderRadius: BorderRadius.circular(widget.radius ?? 20.0)),
        child: TextField(
          controller: widget.controller,
          decoration: InputDecoration(
              hintText: widget.text,
              hintStyle: TextStyle(
                  color: widget.fontColor ?? Colors.black,
                  fontSize: widget.fontSize ?? 18,
                  fontWeight: widget.fontWeight ?? FontWeight.bold),
              enabledBorder: const OutlineInputBorder(
                  borderSide: BorderSide(color: Colors.transparent)),
              focusedBorder: const OutlineInputBorder(
                  borderSide: BorderSide(color: Colors.transparent))),
        ),
      ),
    );
  }
}


lib/example/main.dart

import 'package:flutter/material.dart';
import 'package:gradient_textfield/gradient_textfield.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  TextEditingController email = TextEditingController();
  TextEditingController password = TextEditingController();
  TextEditingController cmpassword = TextEditingController();
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: const Text("Custom Text Field Package"),
        ),
        body: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Gradienttextfield(
              controller: email,
              radius: 40,
              height: 60,
              width: 400,
              colors: const [Colors.grey, Colors.white],
              text: "Email",
              fontColor: Colors.black,
              fontSize: 15,
              fontWeight: FontWeight.normal,
            ),
            const SizedBox(
              height: 20,
            ),
            Gradienttextfield(
                controller: password,
                radius: 40,
                height: 60,
                width: 400,
                colors: const [Colors.pink, Colors.purple],
                text: "Password",
                fontColor: Colors.white,
                fontSize: 15,
                fontWeight: FontWeight.normal),
            const SizedBox(
              height: 20,
            ),
            Gradienttextfield(
                controller: cmpassword,
                radius: 40,
                height: 60,
                width: 400,
                colors: const [Colors.green, Colors.orange],
                text: "Confirm Password",
                fontColor: Colors.white,
                fontSize: 15,
                fontWeight: FontWeight.normal),
          ],
        ));
  }
}
